consensus/spos/bls/subroundBlock.go

```diff
@@ -92,11 +92,13 @@ func (sr *subroundBlock) doBlockJob() bool {
 		return false
 	}
 
-	sentWithSuccess := sr.sendBlock(body, header)
+	sentWithSuccess := sr.sendBlock(header, body)
 	if !sentWithSuccess {
 		return false
 	}
 
+	sr.ConsensusCoreHandler.ScheduledProcessor().StartScheduledProcessing(header, body)
```

Comment:
1. Yes it starts the processing and it has an allotted time (and gas) for finishing (same flow we have for regular txs processing which have a bounded time for execution)
2. The scheduled miniblocks can also be cross shard with the current shard as destination (probably in the follow up PR these need to be treated as well as I believe there were some TODOS), as long as the miniblock contains only smart contract calls (currently this is one limitation but we might decide afterwards if we want to keep this or allow also regular txs).
3. In the case where something fails (e.g implementation bug, not enough time due to wrong gas estimation) then the proposer will not send the final data, so consensus will fail. In case it fails on the validator side but a malicious proposer sends the final info, then the decision will be on the next proposer, which is the scheduled root hash it starts with and which is the previous block it links to.
4. The routine does the actual execution of the already proposed scheduled transactions, so if validators sent their signatures, all of them agreed that they will process. It is the same with cross shard smart contract miniblocks that need to be executed on a different shard, on the source shard it was agreed that the miniblock contains the set of transactions that MUST be executed in that specific order on destination shard. Here as well it is a commitment for execution (but in the same round and on the same shard).